app/models/gallery.rb

```diff
@@ -1,7 +1,8 @@
 class Gallery < ActiveRecord::Base
   belongs_to :user
-  belongs_to :cover_icon, :class_name => Icon
+  belongs_to :cover_icon, class_name: Icon
   has_and_belongs_to_many :icons, after_add: :set_has_gallery, after_remove: :unset_has_gallery
+  has_and_belongs_to_many :ordered_icons, after_add: :set_has_gallery, after_remove: :unset_has_gallery, order: 'LOWER(keyword)', class_name: Icon
```

Comment:
I think this should be an actual scope, rather than a has_*? It feels weird to redefine this relationship.